### PR TITLE
Fix ReST syntax error in the PEP 517 link

### PR DESCRIPTION
**Pull request by mgedmin**
I noticed a ReST formatting problem on the quickstart page:

Ekrano nuotrauka iš 2020-10-05 09-54-05

### CODE REVIEW

**Comment by imba-tjd** [comment]
In this file

```
``twine <https://pypi.org/project/twine/>``
```

has the same issue.

**Comment by mgedmin** [comment]
Fixed the Twine link too, thank you.

File: docs/userguide/quickstart.rst [hunk 1]
```diff
@@ -22,7 +22,7 @@ be generated with whatever tools that provides a ``build sdist``-alike
 functionality. While this may appear cumbersome, given the added pieces,
 it in fact tremendously enhances the portability of your package. The
 change is driven under `PEP 517 <https://www.python.org/dev/peps/pep-0517/#
-build-requirements>``. To learn more about Python packaging in general,
+build-requirements>`_. To learn more about Python packaging in general,
```
Comment (webknjaz) [inline docs/userguide/quickstart.rst:25]: Why not use the ```:pep:``` role instead?
```suggestion
change is driven under :pep:`PEP 517 <517#build-requirements>`_.
To learn more about Python packaging in general,
```

Comment (mgedmin) [reply docs/userguide/quickstart.rst:25]: Sounds good!  GitHub won't let me commit this suggestion for some reason ("Applying suggestions on deleted lines is not supported").

Comment (webknjaz) [reply docs/userguide/quickstart.rst:25]: Yeah... They have this limitation for some reason. But feel free to apply it manually and add a
```
Co-Authored-By: Sviatoslav Sydorenko <webknjaz@redhat.com>
```
line to the commit message.

Comment (mgedmin) [reply docs/userguide/quickstart.rst:25]: I have the `[x] Allow edits by maintainers` checkbox checked, so you could push it too ;)

Comment (webknjaz) [reply docs/userguide/quickstart.rst:25]: Except I'm not a maintainer here. GH doesn't allow to distinguish folks who have write access vs just org members.

File: docs/userguide/quickstart.rst [hunk 2]
```diff
@@ -21,8 +21,8 @@ the backend (build system) it wants to use. The distribution can then
 be generated with whatever tools that provides a ``build sdist``-alike
 functionality. While this may appear cumbersome, given the added pieces,
 it in fact tremendously enhances the portability of your package. The
-change is driven under `PEP 517 <https://www.python.org/dev/peps/pep-0517/#
-build-requirements>``. To learn more about Python packaging in general,
+change is driven under :pep:`PEP 517 <517#build-requirements>`_.
```
Comment (webknjaz) [inline docs/userguide/quickstart.rst:24]: For roles, there shouldn't be a trailing underscores 
```suggestion
change is driven under :pep:`PEP 517 <517#build-requirements>`.
```

**Comment by webknjaz** [comment]
@mgedmin I've made a huge PR recently (#2427) and this one may be unnecessary or needs rebasing.

**Comment by mgedmin** [comment]
https://setuptools.readthedocs.io/en/latest/userguide/quickstart.html today reads

> The change is driven under 517.

and, uh, I think that should be "PEP 517".

**Comment by webknjaz** [comment]
@mgedmin ah, right... The anchor is just `517`. Do you want to send a patch?

**Comment by mgedmin** [comment]
Okay: https://github.com/pypa/setuptools/pull/2439

**Pull request merged**